### PR TITLE
swagger doc for import OAS

### PR DESCRIPTION
**Pull request by urbanishimwe**
updating swagger doc to include `POST:/tyk/import/oas`

## Related Issue
https://tyktech.atlassian.net/browse/TT-5464

### CODE REVIEW

**Review by jeffy-mathew — APPROVED** [review]

**Comment by letzya** [comment]
We need the docs team to review the wording.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4072/merge
Commit: 49cb7a56303357ad948ae7a56368901661705504 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2385140231)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4072/merge
Commit: 30d2d110fa88fcb0bf0ad1328de44b05e619c087 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2392225875)

File: swagger.yml [hunk 1]
```diff
@@ -560,24 +560,9 @@ paths:
...
-            type: string
-        - name: customDomain
-          in: query
-          description: custom domain for the API
-          required: false
-          schema:
-            type: string
+        - $ref: '#/components/parameters/UpstreamURL'
+        - $ref: '#/components/parameters/ListenPath'
+        - $ref: '#/components/parameters/CustomDomain'
```
Comment (jeffy-mathew) [inline swagger.yml:565]: since we have added `allowList`, `validateQuery`, can we add that too?

File: swagger.yml [hunk 2]
```diff
@@ -312,7 +312,7 @@ paths:
   '/tyk/apis/oas':
     get:
       description: |-
-         List APIs, in OAS format (only APIs saved in OAS format)
+         List APIs, in OAS format
```
Comment (jeffy-mathew) [inline swagger.yml:315]: I think this part is important `(only APIs saved in OAS format)` - suppose we have 10 APIs saved and only 30 of them are in OAS format(migrated/created), only 30 APIs will be listed with this endpoint

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4072/merge
Commit: d78b9e6cbd6e8736e8961e382c96ac23bef0d2d0 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2394963789)

**Review by lghiur — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4072/merge
Commit: f71a690f6e1215471880db68403301c7f37491ea 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2408253538)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4072/merge
Commit: 4db7aaa37f8a5812935c1c774493b3901ae25613 
Triggered by: pull_request (@marksou)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2409161980)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/4072/merge
Commit: b2e7bd21006bb62203f036889eaea2677e859e5d 
Triggered by: pull_request (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2413631874)

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/heads/master
Commit: b3c105fb5874d162e1a188f341e38e3015d0cce3 swagger doc for import OAS (#4072)

* swagger doc for import OAS

* capitalize desc

* docs for api ID query

* docs for PATCH OAS

* fix errors in all OAS routes

* Update swagger.yml

Amended some descriptions

Co-authored-by: Mark Southee <mark.southee@gmail.com>
Triggered by: push (@urbanishimwe)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/2413699019)

**Pull request merged**